src/change_stream.ts

```diff
@@ -575,22 +584,29 @@ function waitForTopologyConnected(
   }, 500); // this is an arbitrary wait time to allow SDAM to transition
 }
 
-function closeWithError(changeStream: ChangeStream, error: AnyError, callback?: Callback): void {
+function closeWithError(
+  changeStream: ChangeStream<any>,
```

Comment:
Since this function doesn't need to pipe through the generic (it doesn't use any APIs that depend on what the Schema is) then we can put `any` here. ChangeStream isn't defined to have a default value for the schema since it must be constructed via the `watch()` methods, which do default the generic value. However if we did make `ChangeStream<TSchema = Document>` then these functions would still need to override that with `any` or pass through the generic TSchema despite their implementation being agnostic of the provided generic type.

Reply:
As a general point for the `changeStream` typings, what do you think of adding either a second (optional) generic that users could pass in or a different type that users could customize? ChangeStream documents can vary in shape pretty wildly and I have in fact ended up having to do `changeStream as any` in the past because it conflicted with some of the expected object types, but I could have provided a concrete type instead: Typescript just didn't play nice because it read the types I was trying to convert between as conflicting.

Reply:
It looks like the `ChangeStream` is currently coded to accept a generic that will apply to things like the `fullDocument` returned from the change stream update event, did you have something different in mind? We are also currently looking at typing more of the change stream methods (including this `closeWithError` one which should really be a class method but due to legacy implementation is stuck being a helper method, making strong typing somewhat awkward).

Reply:
I was thinking the `<any>` could be replaced with `<TChangeStream>` and the function can be initialized with `<TChangeStream = any>` for tighter control when needed

Reply:
I believe we have the types covered now, let us know if you see anything else, thanks!

Reply:
I believe we have the types covered now, let us know if you see anything else, thanks!